lib/beaker/test_suite.rb

```diff
@@ -172,7 +172,9 @@ def write_junit_xml(xml_file)
             end
             suite.add_child(properties)
 
-            @test_cases.each do |test|
+            test_cases_to_report = @test_cases
+            test_cases_to_report = @test_cases.sort { |x,y| y.runtime <=> x.runtime } if time_sort
```

Comment:
@waynr why would that be better?  The way that `test_cases_to_report` is currently setup, that would in-place sort `@test_cases` as well, which is why we've done this, to make sure we don't sort that member variable.

The logic for the 2 above lines is that if we're not sorting based on time, then we just use a reference to `@test_cases`, because everything past these lines is read-only, so there's no worry about munging that up.  If time sorting is required, then we want to make a copy, and have that copy be sorted, which is what this line is doing.

Let me know if any of this is unclear, or if you have any questions about it.

Thanks,
Kevin